### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by Navarro-Jonathan**
_**Integration test improvements**_
These changes will test all commands through the CLI. Before, the integration tests were mostly smoke testing the program using a few different commands.

- Added tests for all commands and sub-commands for models, plans, configurations, constraints, expansions, metadata, scheduling, and constraints.
- Added tests for commands that download files
- Removed /'s in paths to support multiple platforms
- Improved simulation tests--we now use "real" configs and run additional simulations after updating the plans.
- Improved documentation
- Improved test structure--split tests into multiple modules, moved session handling into `conftest.py`.

### CODE REVIEW

**Comment by cartermak** [comment]
Closes #21

**Review by cartermak — CHANGES_REQUESTED** [review]
Looks good other than a couple notes. I'm tracking ideas for possible improvements in #62

File: tests/integration_tests/test_plans.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+def test_plans_duplicate():
+    result = runner.invoke(
+        app,
+        ["-c", "localhost", "--hasura-admin-secret", HASURA_ADMIN_SECRET, "plans", "duplicate"],
+        input=str(plan_id) + "\n" + "duplicated_plan" + "\n",
+        catch_exceptions=False,)
+    assert result.exit_code == 0,\
+        f"{result.stdout}"\
+        f"{result.stderr}"
+    assert "Duplicate activity plan created" in result.stdout
```
[the remark this answers is not in the document]
Comment (Navarro-Jonathan) [reply tests/integration_tests/test_plans.py]: I've removed the duplicate duplicate in [5ad9dab](https://github.com/NASA-AMMOS/aerie-cli/pull/66/commits/5ad9dab69265bdab06463e9fd071e6cb22698445)

File: tests/integration_tests/test_scheduling.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    assert result.exit_code == 0,\
+        f"{result.stdout}"\
+        f"{result.stderr}"
+    assert "Assigned goals in priority order" in result.stdout
+
+def test_schedule_delete():
+    result = runner.invoke(
+        app,
+        ["-c", "localhost", "--hasura-admin-secret", HASURA_ADMIN_SECRET, "scheduling", "delete"],
+        input=str(model_id) + "\n" + str(sim_id) + "\n" + GOAL_PATH + "\n",
```
[the remark this answers is not in the document]
Comment (Navarro-Jonathan) [reply tests/integration_tests/test_scheduling.py]: This has now been changed in [62beee6](https://github.com/NASA-AMMOS/aerie-cli/pull/66/commits/62beee6e518ee8cacc70f70cdc6c34a45a6c2a5a)

**Review by cartermak — APPROVED** [review]

**Pull request merged**